### DISCUSSION TITLE
unsupported Scala 2.13.8

### DISCUSSION THREAD

**Discussion by arigoldx** [scalameta/metals#3680] [Q&A] [answered]
Hi there,

I've searched far and wide and tried [the install docs on scalameta.org](https://scalameta.org/metals/docs/editors/emacs) as well as [the install docs on emacs-lsp](https://emacs-lsp.github.io/lsp-mode/page/installation/) and no dice.

I notice `WARN  unsupported Scala 2.13.8` in lsp-log even tho [the docs say that 2.13.8 is supported](https://scalameta.org/metals/docs/editors/emacs).

I know that there are a lot of parts to this and I'm new to using this tool so I could very well be missing something.

Any ideas?

**Comment by ckipp01** [comment, marked as answer]
Hey @arigoldx this is more than likely due to usage of an old Metals version. Which version of Metals are you using, and can you ensure you're using the latest?

**Reply by tanishiking** [reply]
FYI @arigoldx Looks like the metals version can be found from `lsp-metals-install-version` setting :)
https://github.com/emacs-lsp/lsp-metals/blob/5891149014308e73faf025bf40cd099cc185277b/lsp-metals.el#L95-L99

If you set anything to `lsp-metals-install-version`, `emacs-lsp` install the latest version of Metals.

**Reply by arigoldx** [reply]
Thanks!

I just checked and, well, looks like the latest:

```
lsp-metals-install-version is a variable defined in ‘lsp-metals.el’.
Its value is "latest.release"

  You can customize this variable.


This variable was introduced, or its default value was changed, in
version 1.2 of the lsp-metals package.

Documentation:
Metals version to install.
```

:-/

**Reply by ckipp01** [reply]
Correct `latest.release` _will_ install the latest stable version, but the install needs to be triggered. For example if you install today with `latest.release` you'll get `0.11.1`, but if we release `0.11.2` you won't get it unless you re-trigger the install. You can test this by checking wherever lsp-metals installs Metals and just doing a `./metals --version` or a `metals --version` if it installs it on your path. You'll want to ensure you see `0.11.2`.

**Reply by arigoldx** [reply]
_Well that was it!_ Thanks :)

Now that I know I need to re-trigger the install, what the mechanism there? Just delete `~/.emacs.d/.cache/lsp/metals`? Seems like there's gotta be a better way. I searched [the repo](https://github.com/emacs-lsp/lsp-metals/blob/master/lsp-metals.el) but couldn't find anything. Also did a `lsp-metals-<tab>` and searched through completions. No :game_die::game_die:.

Maybe there's something I'm missing? Or maybe there's something I can help improve?

**Reply by arigoldx** [reply]
Hm.. deleting `~/.emacs.d/.cache/lsp/metals/metals` _seemed_ like it cause a re-install but it keeps getting stuck. Here's my `*Messages*`:

```
LSP :: Download metals started.
LSP :: Starting to download https://git.io/coursier-cli to /Users/arigold/.emacs.d/.cache/lsp/metals/coursier...
Contacting host: git.io:443
```

Going to now try to remove the folder, `~/.emacs.d/.cache/lsp/metals`.

**Reply by arigoldx** [reply]
I let it go a while and saw this in `*Messages*`:

> LSP :: Server metals install process failed with the following error message: (error Failed to verify GPG signature: Signature made by expired key CD1B9D8DC84ECA56 Alexandre Archambault (Travis CI coursier) <alexandre.archambault+traviscicoursier@gmail.com>).

Then [following this suggestion](https://github.com/emacs-lsp/lsp-mode/issues/3006#issuecomment-881440918) I tried installing manually with `M-x lsp-install-server` but isn't working (so far) either -- it just hangs.

I'll try again tomorrow :-/

**Reply by ckipp01** [reply]
ahh, you're hitting on this: https://github.com/coursier/coursier/issues/2346 There are some workarounds there.

**Reply by arigoldx** [reply]
that did it! _thanks!_

**Reply by tanishiking** [reply]
Great detective work ckipp01 :tada:
